### PR TITLE
feat(explorer): show detailed account info based on account type

### PR DESCRIPTION
**Pull request by beeman**
## Description

This PR implements the work to make the account pages in the explorer adapt to their type.

The type of account is detected through the `getAccountType` in the solana-client package. In the future we'll be able to add more types here and handle rendering and logic of certain accounts based on that type.

## Screenshots / Video

image

image

image

image

----

> [!IMPORTANT]
> Add detailed account information display in explorer based on account type using `getAccountType`.
> 
>   - **Behavior**:
>     - `ExplorerFeatureAccountInfoContent` in `explorer-feature-account-info-content.tsx` uses `getAccountType` to render account-specific components.
>     - Handles 'token-account', 'token-mint', 'token-unknown', 'system', and 'system-program' types.
>     - Displays error page for unknown token types.
>   - **Components**:
>     - `ExplorerFeatureAccountInfoTokenAccount` and `ExplorerFeatureAccountInfoTokenMint` added for rendering token account and mint details.
>     - `ExplorerFeatureAccountInfo` in `explorer-feature-account-info.tsx` integrates account type display and content rendering.
>     - `ExplorerFeatureAccountOverview` updated to use `ExplorerFeatureAccountInfo`.
>   - **Utilities**:
>     - `ExplorerUiDetailGrid` added for layout consistency.
>   - **i18n**:
>     - Updated `explorer.json` in `locales/en` and `locales/es` for new labels.
>     - Updated `resources.d.ts` to include new i18n keys.
>   - **Constants**:
>     - Added `SYSTEM_ACCOUNT`, `NATIVE_LOADER`, `BPF_LOADER` to `constants.ts`.
>   - **Tests**:
>     - Added integration tests for `getAccountInfo` and `getAccountType` in `get-account-info.integration.test.ts` and `get-account-type.integration.test.ts`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=samui-build%2Fsamui-wallet&utm_source=github&utm_medium=referral)<sup> for 043680217a2eeb47aa2aa7781088c2fe398d561a. You can [customize](https://app.ellipsis.dev/samui-build/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

## Summary by CodeRabbit

* **New Features**
  * Account view now shows token account details (mint, amount, decimals) and token mint details (supply, mint/freeze authorities)
  * Integrated account fetching with in-card loading state

* **Improvements**
  * Simplified account page — overview-focused layout, tabs removed; transactions load after overview
  * Cleaner, explorer-specific error display

* **Localization**
  * Added English and Spanish labels for account/token fields and capitalized "Lamports"

* **Tests**
  * Added tests covering token info parsing and edge cases

### CODE REVIEW

**Comment by tobeycodes** [comment]
@beeman What's the status of this one?

File: packages/explorer/src/explorer-feature-account-info-token-mint.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <ExplorerUiDetailGrid cols={2}>
+      <ExplorerUiDetailRow label="Supply" value={mintSupply} />
+      <ExplorerUiDetailRow label="Decimals" value={mintDecimals} />
+      <ExplorerUiDetailRow
+        label="Mint Authority"
+        value={mintAuthority ? <ExplorerUiLinkAddress address={mintAuthority} basePath={basePath} /> : 'None'}
+      />
+      <ExplorerUiDetailRow
+        label="Freeze Authority"
+        value={freezeAuthority ? <ExplorerUiLinkAddress address={freezeAuthority} basePath={basePath} /> : 'None'}
```
[the remark this answers is not in the document]
Comment (beeman) [reply packages/explorer/src/explorer-feature-account-info-token-mint.tsx]: Fixed by localizing the token mint labels and empty authority fallback through the explorer i18n namespace.

File: packages/explorer/src/explorer-feature-account-info.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}: {
+  basePath: string
+  account: FetchedAccount | undefined
+}) {
+  const { t } = useTranslation('explorer')
+  const type: AccountType = getAccountType({ account })
+  return (
+    <ExplorerUiDetailGrid>
+      <ExplorerUiDetailGrid cols={2}>
+        <ExplorerUiDetailRow label="Account Type" value={<Badge variant="outline">{type}</Badge>} />
```
[the remark this answers is not in the document]
Comment (beeman) [reply packages/explorer/src/explorer-feature-account-info.tsx]: Fixed by replacing the hardcoded Account Type label with an explorer i18n key.

File: packages/explorer/src/explorer-feature-account-overview.tsx [hunk 3]
```diff
@@ -25,41 +26,40 @@ export function ExplorerFeatureAccountOverview({
...
+              <ExplorerUiDetailRow label={t(($) => $.lamports)} value={query.data.lamports} />
+            ) : null}
+          </ExplorerUiDetailGrid>
+
+          {query.isLoading ? <UiLoader /> : <ExplorerFeatureAccountInfo account={query.data} basePath={basePath} />}
+        </ExplorerUiDetailGrid>
+      </UiCard>
+      {query.isError ? (
+        <ExplorerUiError message={query.error.message} title="Error getting account overview" />
+      ) : (
```
[the remark this answers is not in the document]
Comment (beeman) [reply packages/explorer/src/explorer-feature-account-overview.tsx]: Fixed by preventing account details from rendering while the fetch query is in an error state.

File: packages/explorer/src/explorer-feature-account-overview.tsx [hunk 4]
```diff
@@ -25,41 +26,44 @@ export function ExplorerFeatureAccountOverview({
...
+
+          {query.isLoading ? (
+            <UiLoader />
+          ) : query.isError ? null : (
+            <ExplorerFeatureAccountInfo account={query.data} basePath={basePath} />
+          )}
+        </ExplorerUiDetailGrid>
+      </UiCard>
+      {query.isError ? (
+        <ExplorerUiError message={query.error.message} title="Error getting account overview" />
```
[the remark this answers is not in the document]
Comment (beeman) [reply packages/explorer/src/explorer-feature-account-overview.tsx]: Fixed in 68e644a by routing the account overview error title through the explorer i18n namespace and adding the matching English, Spanish, and typed resource entries.

File: packages/explorer/src/explorer-feature-account-overview.tsx [hunk 5]
```diff
@@ -25,41 +26,44 @@ export function ExplorerFeatureAccountOverview({
...
+            {query.data?.exists ? (
+              <ExplorerUiDetailRow label={t(($) => $.lamports)} value={query.data.lamports} />
+            ) : null}
+          </ExplorerUiDetailGrid>
+
+          {query.isLoading ? (
+            <UiLoader />
+          ) : query.isError ? null : (
+            <ExplorerFeatureAccountInfo account={query.data} basePath={basePath} />
+          )}
```
[the remark this answers is not in the document]
Comment (beeman) [reply packages/explorer/src/explorer-feature-account-overview.tsx]: Fixed in 3f809eb by also gating the lamports row on !query.isError, so stale cached account data is hidden whenever the error panel is shown.

File: packages/explorer/src/explorer-feature-account-info-content.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  basePath: string
+  account: ExistingFetchedAccount
+}) {
+  const type: AccountType = getAccountType({ account })
+  const tokenInfo = getAccountTokenInfo({ account })
+
+  switch (type) {
+    case 'token-account':
+      if (tokenInfo?.type !== 'token-account') {
+        return <ExplorerUiError message="Unexpected parsed token account data" title="Unknown token type" />
```
[the remark this answers is not in the document]
Comment (beeman) [reply packages/explorer/src/explorer-feature-account-info-content.tsx]: Fixed in 4d9f626 by routing the fallback token/account error copy through the explorer i18n namespace and adding the matching English, Spanish, and typed resource entries.
